### PR TITLE
Preserve internal navigation indicator on redirect

### PR DESCRIPTION
**Pull request by benhalpern**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This fixes the "double header" for stories path redirects after a user updates their username, etc. This is the same issue as here:

https://github.com/forem/forem/issues/11488

But a different variation of the same bug. This is a specific solution for the stories controller issues. I believe it's localized enough to be worth the quick fix. If we see the same `?i=i` problem elsewhere on the site we could reach for a more general solution.

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Added to documentation?

- [ ] [Developer Docs](https://docs.forem.com) and/or [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [x] No documentation needed

window smush

### CODE REVIEW

File: app/controllers/stories_controller.rb [hunk 1]
```diff
@@ -369,7 +376,7 @@ def assign_listings
   def redirect_to_lowercase_username
     return unless params[:username] && params[:username]&.match?(/[[:upper:]]/)
 
-    redirect_to "/#{params[:username].downcase}", status: :moved_permanently
+    redirect_to "/#{params[:username].downcase}" + internal_nav_param, status: :moved_permanently
```
Comment (vaidehijoshi) [inline app/controllers/stories_controller.rb:379]: could we extract this into a helper? or come up with a way to make it easier for a developer to know when they should be adding the `internal_nav_param` or not? right now, from looking at this, i have no idea when i should append this to a path and when i don't need to, so i have a feeling that it wouldn't be immediately clear when you'd need to include it in a redirected path or not

**Review by vaidehijoshi — COMMENTED** [review]
this seems to work for the _one_ flow that i QAed (change user's username, navigate to old users' old username in the route, expect a redirect). is there anything else i should QA for?

left one question about how we can make it clearer/easier for future devs to know when to include the internal param or not!

File: app/controllers/stories_controller.rb (hunk 1 above)
Comment (rhymes) [reply app/controllers/stories_controller.rb:379]: Agreed, maybe the helper could have a comment explaining what `i=i` is and when to use it. Thanks!

ps. `"#{url}?i=i"` is more idiomatic and faster than `"#{url}" + "?i=i"

Comment (rhymes) [reply app/controllers/stories_controller.rb:379]: Agreed, maybe the helper could have a comment explaining what `i=i` is and when to use it. Thanks!

ps. `"#{url}?i=i"` is more idiomatic and faster than `"#{url}" + "?i=i"

Comment (rhymes) [reply app/controllers/stories_controller.rb:379]: Agreed, maybe the helper could have a comment explaining what `i=i` is and when to use it. Thanks!

ps. `"#{url}?i=i"` is more idiomatic and faster than `"#{url}" + "?i=i"

Comment (rhymes) [inline app/controllers/stories_controller.rb:379]: Agreed, maybe the helper could have a comment explaining what `i=i` is and when to use it. Thanks!

ps. `"#{url}?i=i"` is more idiomatic and faster than `"#{url}" + "?i=i"

**Review by rhymes — COMMENTED** [review]
+1 on the helper suggestion

**Comment by benhalpern** [comment]
@vaidehijoshi @rhymes I extracted this into something we can re-use more often that encapsulates everything we need from it, and added some documentation to explain more about this overall situation and where URL redirects might originate in the app most relevantly.

**Review by rhymes — APPROVED** [review]
Thanks @benhalpern!

LGTM

**Review by atsmith813 — APPROVED** [review]

**Pull request merged**